src/mapping/query/function/ceil.rs

```diff
@@ -0,0 +1,117 @@
+
+#[derive(Debug)]
+pub(in crate::mapping) struct CeilFn {
+    query: Box<dyn Function>,
+    precision: Option<Box<dyn Function>>,
+}
+
+impl CeilFn {
+    #[cfg(test)]
+    pub(in crate::mapping) fn new(
+        query: Box<dyn Function>,
+        precision: Option<Box<dyn Function>>,
+    ) -> Self {
+        Self { query, precision }
+    }
+}
+
+impl Function for CeilFn {
+    fn execute(&self, ctx: &Event) -> Result<Value> {
+        let precision = optional!(ctx, self.precision, Value::Integer(v) => v).unwrap_or(0);
+        let res = required!(ctx, self.query,
+                            Value::Float(f) => {
+                                Value::Float(round_to_precision(f, precision, f64::ceil))
+                            },
+                            v@Value::Integer(_) => v
+        );
+
+        Ok(res)
+    }
+
+    fn parameters() -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                accepts: |v| matches!(v, Value::Float(_) | Value::Integer(_)),
+                required: true,
+            },
+            Parameter {
+                keyword: "precision",
+                accepts: |v| matches!(v, Value::Integer(_)),
+                required: false,
+            },
+        ]
+    }
+}
+
+impl TryFrom<ArgumentList> for CeilFn {
+    type Error = String;
+
+    fn try_from(mut arguments: ArgumentList) -> Result<Self> {
+        let query = arguments.required("value")?;
+        let precision = arguments.optional("precision");
+
+        Ok(Self { query, precision })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::mapping::query::path::Path;
+
+    #[test]
+    fn ceil() {
+        let cases = vec![
+            (
+                Event::from(""),
+                Err("path .foo not found in event".to_string()),
+                CeilFn::new(Box::new(Path::from(vec![vec!["foo"]])), None),
+            ),
+            (
+                Event::from(""),
+                Ok(Value::from(1235.0)),
+                CeilFn::new(Box::new(Literal::from(Value::Float(1234.8))), None),
+            ),
+            (
+                Event::from(""),
+                Ok(Value::from(1234)),
+                CeilFn::new(Box::new(Literal::from(Value::Integer(1234))), None),
+            ),
+            (
+                Event::from(""),
+                Ok(Value::from(1234.4)),
+                CeilFn::new(
+                    Box::new(Literal::from(Value::Float(1234.39429))),
+                    Some(Box::new(Literal::from(Value::from(1)))),
+                ),
+            ),
+            (
+                Event::from(""),
+                Ok(Value::from(3.1416)),
+                CeilFn::new(
+                    Box::new(Literal::from(Value::Float(std::f64::consts::PI))),
+                    Some(Box::new(Literal::from(Value::from(4)))),
+                ),
+            ),
+            (
+                Event::from(""),
```

Comment:
Nit: I realize this existed in the other PRs I've reviewed too, but it seems like we could drop `Event::from("")` from these tuples since it is always the same and just create one in the below loop.

Reply:
This is the same pattern that every remap function follows, so I think it is best to keep it there. But, I think the issue is that it is always the same. There should be at least one test that tests that the function can work with a field coming from the event. I'll add some.